pyuploadcare/dj/conf.py

```diff
@@ -24,25 +24,70 @@
-    "widget_build", settings.UPLOADCARE.get("widget_variant", "full.min")
-)
+legacy_widget = settings.UPLOADCARE.get("legacy_widget", False)
+
+use_hosted_assets = settings.UPLOADCARE.get("use_hosted_assets", True)
 
-widget_filename = "uploadcare.{0}.js".format(widget_build).replace("..", ".")
+# Legacy widget (uploadcare.js)
 
-hosted_url = "https://ucarecdn.com/libs/widget/{version}/{filename}".format(
-    version=widget_version, filename=widget_filename
+legacy_widget_version = settings.UPLOADCARE.get("legacy_widget_version", "3.x")
+legacy_widget_build = settings.UPLOADCARE.get(
+    "legacy_widget_build",
+    settings.UPLOADCARE.get("legacy_widget_variant", "full.min"),
+)
+legacy_widget_filename = "uploadcare.{0}.js".format(
+    legacy_widget_build
+).replace("..", ".")
+legacy_hosted_url = (
+    "https://ucarecdn.com/libs/widget/{version}/{filename}".format(
+        version=legacy_widget_version, filename=legacy_widget_filename
+    )
 )
 
-local_url = "uploadcare/{filename}".format(filename=widget_filename)
+legacy_local_url = "uploadcare/{filename}".format(
+    filename=legacy_widget_filename
+)
+legacy_uploadcare_js = (
+    legacy_hosted_url if use_hosted_assets else legacy_local_url
+)
 
-use_hosted_assets = settings.UPLOADCARE.get("use_hosted_assets", True)
+if "legacy_widget_url" in settings.UPLOADCARE:
+    legacy_uploadcare_js = settings.UPLOADCARE["legacy_widget_url"]
+
+# New widget (blocks.js)
+
+widget_options = settings.UPLOADCARE.get("widget_options", {})
+
+widget_version = settings.UPLOADCARE.get("widget_version", "0.26.0")
+widget_build = settings.UPLOADCARE.get(
+    "widget_build", settings.UPLOADCARE.get("widget_variant", "regular")
+)
+widget_filename_js = "blocks.min.js"
+widget_filename_css = "lr-file-uploader-{0}.min.css".format(widget_build)
+hosted_url_js = "https://cdn.jsdelivr.net/npm/@uploadcare/blocks@{version}/web/{filename}".format(
+    version=widget_version, filename=widget_filename_js
+)
+hosted_url_css = "https://cdn.jsdelivr.net/npm/@uploadcare/blocks@{version}/web/{filename}".format(
+    version=widget_version, filename=widget_filename_css
+)
+
+local_url_js = "uploadcare/{filename}".format(filename=widget_filename_js)
+local_url_css = "uploadcare/{filename}".format(filename=widget_filename_css)
 
 if use_hosted_assets:
-    uploadcare_js = hosted_url
+    uploadcare_js = hosted_url_js
+    uploadcare_css = hosted_url_css
 else:
-    uploadcare_js = settings.UPLOADCARE.get("widget_url", local_url)
+    uploadcare_js = settings.UPLOADCARE.get(
+        "widget_url_js", settings.UPLOADCARE.get("widget_url", local_url_js)
+    )
+    uploadcare_css = settings.UPLOADCARE.get("widget_url_css", local_url_css)
 
-upload_base_url = settings.UPLOADCARE.get("upload_base_url")
+
+if "widget_url_js" in settings.UPLOADCARE:
+    uploadcare_js = settings.UPLOADCARE["widget_url_js"]
+
+if "widget_url_css" in settings.UPLOADCARE:
+    uploadcare_css = settings.UPLOADCARE["widget_url_css"]
```

Comment:
this is confusing
can we simplify this?

Reply:
@dmitry-mukhin I will think about it. In a way, it became less confusing than what we've had before.

Reply:
ok

Reply:
Probably should scrap this. It's not very flexible.

Reply:
@dmitry-mukhin updated the whole `pyuploadcare.dj.conf` module.